simulariumio/writers/writer.py

```diff
@@ -265,6 +268,17 @@ def _check_agent_ids_are_unique_per_frame(buffer_data: Dict[str, Any]) -> bool:
                 get_n_subpoints = True
         return True
 
+    @staticmethod
+    def _validate_ids(trajectory_data: TrajectoryData) -> None:
+        """
+        Check if agent unique IDs are valid 32 bit integers
+        returns a message identifying violating agent ID
+        """
+        agent_unique_ids = trajectory_data.agent_data.unique_ids
+        for uid in np.ndarray.flatten(agent_unique_ids):
+            if uid > MAX_AGENT_ID:
+                raise DataError(f"Agent IDs is larger than a 32 bit integer: {uid} ")
```

Comment:
Might be useful for the user to know which agent and timestep the bad ID is at, although it would make this more slow to loop through the multi-dimensional array. Maybe there's a way to use slicing?